cmd/crossplane/core/core.go

```diff
@@ -98,7 +99,9 @@ type startCommand struct {
 
 	XpkgCacheDir string `aliases:"cache-dir" default:"/cache/xpkg" env:"XPKG_CACHE_DIR,CACHE_DIR" help:"Directory used for caching package images." short:"c"`
 
-	PackageRuntime string `default:"Deployment" env:"PACKAGE_RUNTIME" help:"The package runtime to use for packages with a runtime (e.g. Providers and Functions)"`
+	PackageRuntime         string `default:"Deployment" env:"PACKAGE_RUNTIME"          help:"The package runtime to use for packages with a runtime (e.g. Providers and Functions)"`
+	PackageFunctionRuntime string `default:"Deployment" env:"PACKAGE_FUNCTION_RUNTIME" help:"The package runtime to use for Function packages"`
+	PackageProviderRuntime string `default:"Deployment" env:"PACKAGE_PROVIDER_RUNTIME" help:"The package runtime to use for Provider packages"`
```

Comment:
Instead of controlling this with 3 flags (and more if we add new package types in future) we could encode this in the value of `--package-runtime`. For example with a syntax like `[package-type=runtime-type,...]`:

* `--package-runtime Deployment` - Use Deployment everywhere
* `--package-runtime 'Provider=Deployment,Function=External'` - Specify granularly

I like that this'd use fewer flags, and remove any erroneous interactions between flags. I don't love that it requires you to get syntax right in the string, but that's not super uncommon for CLI flag values.

Reply:
Alternatively, can we drop `--package-runtime` and move to only `--function-runtime` and `--provider-runtime`?

If we really care about not breaking folks, we could make `--package-runtime` a hidden flag that sets both the new flags if you specify it. That said, I'd be really surprised if anyone was setting `--package-runtime` in the wild today so I'm not really worried about changing this flag breaking anyone.

Reply:
I wouldn't worry much about breaking:
- Will be first released in v2
- I don't think anyone uses it yet.